pkg/admin/migration/types.go

```diff
@@ -0,0 +1,70 @@
+package migration
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type MigratonStatus string
+
+const (
+	MigrationSuccess MigratonStatus = "success"
+	MigrationFailure MigratonStatus = "failure"
+)
+
+// Container for individual migration attempts
+type MigrationResult struct {
+	Error           error                       `json:"error,omitempty"`
+	ResourceVersion schema.GroupVersionResource `json:"resourceVersion"`
+	Timestamp       time.Time                   `json:"timestamp"`
+	ObjectMeta      *metav1.ObjectMeta          `json:"object,omitempty"`
+}
+
+type MigrationResultList struct {
```

Comment:
Not sure if we need that. Is there a way to log that some resource was migration from version A to B? We could log that in "real time", not packing all results into a collection and then iterating to log the items

Reply:
Sure thing, we can do that. The primary reason it's here is just to give us raw access to write the information in any format we want if we need to store the results somewhere easily.

Reply:
Is the pre-run phase logging to the standard MicroShift log, or do we see these errors in the greenboot health check log? It would be nice if the greenboot log could at least report "there was a data migration error, check the MicroShift logs for details".

Reply:
It's currently just using klog, I'm not sure how we send logs to greenboot but I love the idea. We should make it do that

Reply:
The klog output is going to go to stdout/stderr. I guess that's going to the systemd unit where pre-run is running, rather than a greenboot-specific log. Can we put these messages (and any others) somewhere for our health-check script to collect and report?

Reply:
Sure, I added some files here to use in greenboot, https://github.com/openshift/microshift/pull/1956/files#diff-e78e17834c76663b756393d04ebe7aceb8ea4059d44b7d2ae3ee4e9999aead6eR23-R24